GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/GVFSUpgradeNagTests.cs

```diff
@@ -0,0 +1,87 @@
+﻿using GVFS.FunctionalTests.FileSystemRunners;
+using GVFS.FunctionalTests.Tools;
+using GVFS.Tests.Should;
+using NUnit.Framework;
+using System;
+using System.IO;
+using System.Linq;
+
+namespace GVFS.FunctionalTests.Tests.EnlistmentPerFixture
+{
+    [TestFixture]
+    [Category(Categories.FullSuiteOnly)]
+    [Category(Categories.Windows)]
+    public class UpgradeNagTests : TestsWithEnlistmentPerFixture
+    {
+        private const string GVFSInstallerName = "SetupGVFS.1.0.18234.1.exe";
+        private const string GitInstallerName = "Git-2.17.1.gvfs.2.5.g2962052-64-bit.exe";
+
+        private string upgradeDirectory;
+        private FileSystemRunner fileSystem;
+
+        public UpgradeNagTests()
+        {
+            this.fileSystem = new SystemIORunner();
+            this.upgradeDirectory = Path.Combine(
+                Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData, Environment.SpecialFolderOption.Create),
+                "GVFS",
+                "GVFS.Upgrade",
+                "Downloads");
+        }
+
+        [TestCase]
+        public void NoNagWhenUpgradeNotAvailable()
+        {
+            this.EmptyDownloadDirectory();
+
+            ProcessResult result = GitHelpers.InvokeGitAgainstGVFSRepo(
+                this.Enlistment.RepoRoot,
+                "status");
+
+            string.IsNullOrEmpty(result.Errors).ShouldBeTrue();
+        }
+
+        [TestCase]
+        public void NagWhenUpgradeAvailable()

```

Comment:
You might also consider adding a test that puts randomly named  files in `GVFS.Upgrade\Downloads` and confirms that the nag is (or is not) displayed.